### ISSUE TITLE
Weird / inconsistent deep paging behavior

### ISSUE THREAD

**Issue by eddotman** [CrossRef/rest-api-doc#294]
**tl;dr** Deep paging `/works/` w/ proper format and generous timeouts between cursors (~1.5s) gives inconsistent behavior and returns less-than-max results.

Using the following code which retrieves DOIs from some list of ISSNs...

```python
import requests
import time

min_year = 1995
issns = [
    "0002-7863",
]
all_dois = set()

for issn in issns:
    cursor = "*"
    keep_paging = True
    while (keep_paging):
        time.sleep(1.5)
        r = requests.get("http://api.crossref.org/journals/" + issn +
                               "/works/?filter=from-pub-date:" +
                               str(min_year) + "&rows=1000&cursor=" + cursor)
        cursor = requests.utils.quote(r.json()['message']['next-cursor'], safe='')
        if len(r.json()['message']['items']) == 0:
            keep_paging = False

        for item in r.json()['message']['items']:
            doi = item['DOI']
            all_dois.add(doi)
        
        print issn + ' | #DOIS: ' + str(len(all_dois))
```

I find the following output:

```
0002-7863 | #DOIS: 1000
0002-7863 | #DOIS: 2000
0002-7863 | #DOIS: 3000
0002-7863 | #DOIS: 4000
0002-7863 | #DOIS: 5000
0002-7863 | #DOIS: 5998
0002-7863 | #DOIS: 6962
0002-7863 | #DOIS: 7955
0002-7863 | #DOIS: 8945
0002-7863 | #DOIS: 9897
0002-7863 | #DOIS: 10855
0002-7863 | #DOIS: 11801
0002-7863 | #DOIS: 12733
0002-7863 | #DOIS: 13644
0002-7863 | #DOIS: 14559
0002-7863 | #DOIS: 14697
0002-7863 | #DOIS: 14699
0002-7863 | #DOIS: 14699
```

In particular, the following points seem strange:

- Queries in the middle return less than the requested rows before hitting the end of the results set
- Running the same query returns an inconsistent number of results
- (Corollary:) The end of the results set doesn't seem to be the actual end, especially when comparing to the metadata record for max DOIs retrieved when doing a single `&rows=0` query.

Also, increasing / decreasing sleep times between cursors didn't seem to have a consistent effect.

Anyway, any help on this issue would be great. Thanks!

**Comment by kjw** [comment]
Worth noting that the last token received can be used to tail a result set. That said, it would be pretty odd for ~5,000 records to appear for this ISSN while you were retriving the result set. 

I'm suspicious that this could be the result of us running two different indexes in production over the last few days that were slightly out of sync with each other. Can I ask when you ran this?

**Comment by eddotman** [comment]
I was running this today / yesterday. Maybe that's the issue? I can try again later this week, but I've replicated the issue with several other ISSNs too.

**Comment by eddotman** [comment]
@kjw Yeah good call - just ran it again now and I'm seeing the correct output. Small snippet follows:

```
0028-0836 | #DOIS: 42000
0028-0836 | #DOIS: 43000
0028-0836 | #DOIS: 44000
0028-0836 | #DOIS: 45000
0028-0836 | #DOIS: 46000
0028-0836 | #DOIS: 47000
0028-0836 | #DOIS: 48000
```

So even for very large paged queries, I see no issues. Thanks!
